plugin-script/src/main/java/io/kestra/plugin/scripts/exec/AbstractExecScript.java

```diff
@@ -0,0 +1,88 @@
+package io.kestra.plugin.scripts.exec;
+
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.models.tasks.Task;
+import io.kestra.core.runners.RunContext;
+import io.kestra.plugin.scripts.exec.scripts.models.DockerOptions;
+import io.kestra.plugin.scripts.exec.scripts.models.RunnerType;
+import io.kestra.plugin.scripts.exec.scripts.models.ScriptOutput;
+import io.kestra.plugin.scripts.exec.scripts.runners.Commands;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+
+import java.util.List;
+import java.util.Map;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+public abstract class AbstractExecScript extends Task implements RunnableTask<ScriptOutput> {
+    @Builder.Default
+    @Schema(
+        title = "Runner to use"
+    )
+    @PluginProperty
+    @NotNull
+    @NotEmpty
+    protected RunnerType runner = RunnerType.DOCKER;
+
+    @Schema(
+        title = "Docker options when using runner `DOCKER`"
+    )
+    @PluginProperty
+    protected DockerOptions docker;
+
+    @Schema(
+        title = "The commands to run before"
+    )
+    @PluginProperty(dynamic = true)
+    protected List<String> beforeCommands;
+
+    @Schema(
+        title = "Additional environments variable to add for current process."
+    )
+    @PluginProperty(
+        additionalProperties = String.class,
+        dynamic = true
+    )
+    protected Map<String, String> environments;
```

Comment:
I liked the `env`moniker used before. Seems familiar to folks using GitHub Actions, K8s, etc.